### PR TITLE
add tests for showing help

### PR DESCRIPTION
**Pull request by fricklerhandwerk**
based on #10137

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

File: tests/functional/help.sh [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+nix store make-content-addressed --help
+nix store optimise --help
+nix store path-from-hash-part --help
+nix store ping --help
+nix store prefetch-file --help
+nix store repair --help
+nix store sign --help
+nix store verify --help
+nix upgrade-nix --help
+nix why-depends --help
```
Comment (edolstra) [inline tests/functional/help.sh:144]: This list seems like overkill and a pain to maintain. If we really want to test the help of every command, it's better to do something like nix __dump-cli | jq .args.commands ... to extract the list of commands.

Comment (fricklerhandwerk) [reply tests/functional/help.sh:144]: Done. It's quite involved but fits the bill.

File: tests/functional/help.sh [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  else
+    $prefix
+  end;
+
+.args.commands | to_entries[] | .key as $cmd |
+  if .value | has("commands") then
+    (.value | recurse($cmd))
+  else
+    $cmd
+  end
```
Comment (thufschmitt) [inline tests/functional/help.sh:73]:
```suggestion
def recurse($prefix):
    to_entries[] |
    ($prefix + [.key]) as $newPrefix |
    (if .value | has("commands") then
      ($newPrefix, (.value.commands | recurse($newPrefix)))
    else
      $newPrefix
    end);
.args.commands | recurse([]) | join(" ")
```

should be less scary

(and yes, I just spent way too long on that)
```

File: tests/functional/local.mk [hunk 3]
```diff
@@ -129,7 +129,8 @@ nix_tests = \
   read-only-store.sh \
   nested-sandboxing.sh \
   impure-env.sh \
-  debugger.sh
+  debugger.sh \
+	help.sh
```
Comment (thufschmitt) [inline tests/functional/local.mk:133]:
```suggestion
  help.sh
```

**Review by thufschmitt — CHANGES_REQUESTED** [review]

**Pull request merged**